### DISCUSSION TITLE
Allow navigating to sub by exact name

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [Docile-Alligator/Infinity-For-Reddit#1284] [Completed Ideas] [answered]
Sometimes, a particular subreddit will not show up in search results. This can be particularly frustrating, especially when you know the exact name and would be otherwise able to get there. This proposal will add a fallback, a way to go somewhere directly by name, all while reusing as much of the existing UI as possible.

I would like to propose a new meta element in search - if you type a value in the search bar and are looking at the subreddits search results, there should be an entry at the top that matches the exact text you provided. When you tap this entry, it would attempt to navigate directly to that subreddit.

There are a few conditions here that would need to be addressed:

1. The name must match the format for valid subreddit names. This can be done with a simple REGEX query without issue, and if it doesn't match then this meta result should simply be hidden.
2. The searched for subreddit exists in the search results (exact name match). In this case, just hide the meta result. For example, if someone searches "cats," the search entry for `r/cats` should take priority over the generic entry for "go directly to r/cats" as the search entry will have better UX anyways (including the subreddit image and subscriber count).
3. NSFW Subreddits - sometimes, a subreddit will not show up in the search results because it is flagged as NSFW. If the user has NSFW enabled, this should just continue through without an issue. If not, a warning banner should appear to inform the user that the sub the user searched for requires NSFW to be enabled with options to either go to settings or to cancel. If NSFW was permanently disabled, I'm unsure whether to display a similar warning without the option to enable or to just pretend like the sub was not found.
4. Subreddit not found - This also requires an error informing the user the searched for subreddit does not exist. It should include helpful text such as a note that the sub may have been deleted or banned, and if banned, the reason why. If Infinity ever gets the ability to create subreddits, this could be a good way to facilitate this option, as it naturally avoids frustrations with name collisions.
5. Temporary network issues - same as for regular search results. Sometimes timeouts or 500 errors may be returned from the API. Standard error handling should be able to cover these cases but testing should be included in order to ensure this is handled properly.

**Appearance**

As far as how this meta entry should look, I imagine it should appear in the same form as the other entries. The subreddit icon should be either a simplified Infinity or Reddit logo and the sub title should be "Go directly to r/[search]...", perhaps italicized so the entry stands out more. As for the line that normally shows the subscriber count, I propose this line either be left blank or some witty text be provided, though I'm unsure exactly what our options are here based on how the UI is built.

**Comment by us3rname101** [comment, marked as answer]
As far as I can tell, this feature already exists. Long press the FAB (the plus button in the feed) and it should show a selection of actions and one of them is "Go to subreddit". You can a type a sub name and press OK and it'll go there.

**Reply by KaKi87** [reply]
It's still a workaround, one that requires more clicks than the actual feature would need.
Thanks

**Reply by us3rname101** [reply]
Do you understand the definition of workaround?


Thanks

**Reply by KaKi87** [reply]
I do.

**Reply by us3rname101** [reply]
Then you can tell me how the feature is a workaround.

**Reply by KaKi87** [reply]
The requested feature is the ability to navigate by exact match using search when autocompletion & results don't return the desired item.

Meanwhile, the proposed solution requires opening a separate menu, which either makes the user go back and type their query a second time if they tried the search feature first, or makes them type their query directly in full without giving the search feature a try first.

Additionally, it also takes more time to open since it requires a long press and finding it in a list while the search feature requires no time to find and click.

**Reply by us3rname101** [reply]
> The requested feature is the ability to navigate by exact match

That's what it does, no?

> the user go back and type their query a second time

The user is hopefully smart enough to copy the text

> makes them type their query directly in full without giving the search feature a try first.

It shows search results
![Screenshot_2023-06-16-00-00-39-97_2656fe66d4cc08342e93fbdd9fd1d373](https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/95178906/1ab5a07a-7e0b-465c-a23d-e5b776a830ee)

> takes more time ... finding it in a list

You'll get muscle memory'ed if you use it much. It takes like a second to open the pop up, i don't think one second is that relevant.

**Reply by KaKi87** [reply]
> The user is hopefully smart enough to copy the text

This also requires additional clicks, and still requires going back.

> It shows search results

Oh, I was submitting so fast that I didn't notice that, thanks.

> i don't think one second is that relevant

Every added second is bad for UX.

There also are two more advantages in OP's requested feature :
- it could also work for user searching ;
- it features history while menu search features don't.

To conclude, while there are a lot of cases where UX optimization efforts aren't worth the time these save, this one is totally worthy with its all-in-one advantages.

Thanks

**Reply by us3rname101** [reply]
> Every added second is bad for UX.

Does this apply when the button has two click modes :p

> it could also work for user searching

It's your lucky day that button is in the list too, minus the list of users that match the text


While I agree that a button in the normal search button, might be better, I don't think many users would actually use it that much. Like if you search "cats", in the time you are moving your finger to press a button (at the top, next to the clear button) to directly to go to a subreddit it already shows the sub in list.

> it features history while menu search features don't.

If youre going to a same sub multiple times, maybe you should subscribe to it and access it from the Subscriptions :p


It's fine how it is currently, there's little benefit to moving it.

**Reply by KaKi87** [reply]
> Does this apply when the button has two click modes :p

Not for all its features, no. But for search yes, because it **adds** wait time compared to the normal search button.

> that button is in the list too

I saw, but again, exact user match from the search feature would provide the same several advantages it would provide for subreddits.

> Like if you search "cats", in the time you are moving your finger to press a button (at the top, next to the clear button) to directly to go to a subreddit it already shows the sub in list.

I'm sorry I don't understand that sentence.

> If youre going to a same sub multiple times, maybe you should subscribe to it and access it from the Subscriptions :p

That's right, though. ^^

**Reply by us3rname101** [reply]
> adds wait time

Well yes but how else would you distinguish from the buttons two mods, post creation and menu list for action? 


About the user match, the search bar is getting kinda crowded 
![IMG_20230616_103147](https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/95178906/4461c316-265a-42a3-a20f-03c7a091f8ba)


> I'm sorry I don't understand that sentence.

Sorry if that was unclear.

Ok so you are searching for a sub, your fingers are the bottom of the phone on the keyboard, right?

And lets say the hypothetical button to directly go to a sub is with the other buttons (Clear search next and  handle link button).

You stop typing and begin to move your finger to press the button, but it already shows the correct sub in the search list, right?




![IMG_20230616_103802](https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/95178906/f37f20f7-060f-4561-b7d3-0d01decfc421)

**Reply by KaKi87** [reply]
> how else would you distinguish from the buttons two mods, post creation and menu list for action?

In no other way : the current way is fine, I'm just saying that there would be no need to use search features from there thanks to OP's suggestion.

> About the user match, the search bar is getting kinda crowded

OP's suggestion is not to add a button next to the input, but to display a result item that points to the user's exact input.

> it already shows the correct sub in the search list, right?

This feature request is precisely for when it isn't, as OP's first sentence states.

---

Infinity doesn't suck, it is awesome, I can't use Reddit without it, and I recommend it everytime I have an opportunity.

But, here, there's an oppportunity to improve (i.e. to be even more awesome), so I'm asking for it to be taken.

Thanks

**Reply by us3rname101** [reply]
OP's wishes was answered and they were fine it.

**Reply by KaKi87** [reply]
> they were fine

How do you know ?

**Reply by us3rname101** [reply]
They would have replied.

**Reply by KaKi87** [reply]
Turns out [RedReader](https://github.com/QuantumBadger/RedReader), which I discovered today, does that very well, although inferior to Infinity on almost every other aspect.

**Comment by Wladefant** [comment]
As @us3rname101 said this feature already exists, but we will definitely look at your suggestions
